### PR TITLE
DI-222 Introduce Json logging

### PR DESCRIPTION
**Pull request by ed-sparkes**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-222

## Description

Use the aws lambda powertools logger to log JSON and include lambda context

### Noteworthy Changes

- No longer any file logging going on, only stdout logging in JSON format

## Type of change

Delete not appropriate

- New feature (non-breaking change which adds functionality)

## Testing

Please tick the testing that has been completed

- [x] Unit tests
- [x] Component tests

## Developer Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the [code formatting checks](../README.md#code-quality)
- [x] I have run the [code quality checks](../README.md#code-quality)
- [x] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [x] Unit test code coverage is at or above 80%
- [x] New and existing unit tests pass locally with my changes
- [x] Tests have added that prove my fix is effective or that my feature works (Component tests)
- [x] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

File: application/common/utilities.py [hunk 1]
```diff
@@ -58,7 +58,8 @@ def invoke_lambda_function(lambda_name: str, lambda_event: Dict[str, Any]) -> No
         lambda_name (str): Name of lambda to invoke
         lambda_event (Dict[str, Any]): Event to pass to lambda
     """
-    logger.info(f"Invoking {lambda_name}")
+    lambda_event["correlation_id"] = logger.get_correlation_id()
```
Comment (rohith1122) [inline application/common/utilities.py:61]: What is the use of this line? I  don't think we are using it anywhere

File: application/dos_api_gateway/dos_api_gateway.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
 from json import loads, dumps
 from typing import Dict, Any
 from aws_lambda_powertools.utilities.typing import LambdaContext
+from aws_lambda_powertools import Logger
 
+logger = Logger()
 
+
+@logger.inject_lambda_context(correlation_id_path="headers.x_correlation_id")
```
Comment (rohith1122) [inline application/dos_api_gateway/dos_api_gateway.py:9]: How to handle if they don't pass x_correlation_id in headers. do we need to implement any validation?

**Review by JackPlowman — DISMISSED** [review]
LGTM 👍

File: application/dos_api_gateway/dos_api_gateway.py (hunk 2 above)
Comment (ed-sparkes) [reply application/dos_api_gateway/dos_api_gateway.py:9]: its optional, although we should probably check if the correlation id is provided and if not set one so we have the tracking at least ill do this now. if the header is not provided it continues gracefully

**Review by rohith1122 — DISMISSED** [review]
LGTM

File: application/dos_api_gateway/dos_api_gateway.py (hunk 2 above)
Comment (ed-sparkes) [reply application/dos_api_gateway/dos_api_gateway.py:9]: in fact ill raise a subsequent PR for this

**Review by rohith1122 — APPROVED** [review]

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Pull request merged**